wandb/sdk/launch/docker.py

```diff
@@ -64,31 +64,30 @@ def generate_docker_image(project: _project_spec.Project, entry_cmd):
     if not image_id:
         image_id = re.findall(r"Reusing existing image \((.+)\)", stderr)
     if not image_id:
-        raise Exception("error running repo2docker")
+        raise Exception("Error generating docker image")
     return image_id[0]
 
 
-def build_docker_image(project: _project_spec.Project, repository_uri, base_image, api):
+def build_docker_image(project: _project_spec.Project, name, base_image, api):
     """
     Build a docker image containing the project in `work_dir`, using the base image.
     """
 
     image_uri = _get_docker_image_uri(
-        repository_uri=repository_uri, work_dir=project.dir
+        name=name, work_dir=project.dir
     )
 
     wandb_project = project.docker_env["WANDB_PROJECT"]
     wandb_entity = project.docker_env["WANDB_ENTITY"]
     dockerfile = (
         "FROM {imagename}\n"
-        "COPY {build_context_path}/ {workdir}\n"
+        "COPY --chown={uid} {build_context_path}/ {workdir}\n"
```

Comment:
ok so we currently copy in the repo twice (once here and once via r2d). should resolve this

Reply:
Good catch. We should check that repo2docker is putting the code in the right place for the docker argument and potentially drop this one.

This also brings up a related point, when a user specifies a docker image, I think we should assume that image contains the code?

Reply:
according to usage below, it looks like uid might be None. Should we handle that here?